### PR TITLE
Fix principal not available issue when aligning transcripts

### PR DESCRIPTION
**Pull request by zhx828**

### CODE REVIEW

File: src/main/java/org/mskcc/oncokb/curation/security/SecurityUtils.java [hunk 1]
```diff
@@ -44,6 +47,11 @@ private static String extractPrincipal(Authentication authentication) {
             if (attributes.containsKey("preferred_username")) {
                 return (String) attributes.get("preferred_username");
             }
+        } else if (authentication.getPrincipal() instanceof DefaultOAuth2User) {
```
Comment (zhx828) [inline src/main/java/org/mskcc/oncokb/curation/security/SecurityUtils.java:50]: @calvinlu3 what changed recently that we need to add this clause? I handle DefaultOidcUser

File: src/main/java/org/mskcc/oncokb/curation/security/SecurityUtils.java [hunk 2]
```diff
@@ -9,7 +9,10 @@
 import org.springframework.security.core.context.SecurityContext;
 import org.springframework.security.core.context.SecurityContextHolder;
 import org.springframework.security.core.userdetails.UserDetails;
+import org.springframework.security.oauth2.core.OAuth2AccessToken;
```
Comment (calvinlu3) [inline src/main/java/org/mskcc/oncokb/curation/security/SecurityUtils.java:12]:
```suggestion
```

File: src/main/java/org/mskcc/oncokb/curation/security/SecurityUtils.java [hunk 3]
```diff
@@ -9,7 +9,10 @@
...
 import org.springframework.security.oauth2.core.oidc.user.DefaultOidcUser;
+import org.springframework.security.oauth2.core.user.DefaultOAuth2User;
+import org.springframework.security.oauth2.core.user.OAuth2User;
```
Comment (calvinlu3) [inline src/main/java/org/mskcc/oncokb/curation/security/SecurityUtils.java:15]:
```suggestion
```

**Review by calvinlu3 — APPROVED** [review]
LGTM

**Pull request merged**